pkg/cloud/services/network/subnets.go

```diff
@@ -336,6 +343,7 @@ func (s *Service) describeVpcSubnets() (infrav1.Subnets, error) {
 	for _, ec2sn := range sns.Subnets {
 		spec := infrav1.SubnetSpec{
 			ID:               *ec2sn.SubnetId,
```

Comment:
Do we still want this to be set to the subnetId?

Reply:
We need to yeah, otherwise the ID is empty and it gets stored empty